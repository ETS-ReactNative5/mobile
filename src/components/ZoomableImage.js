import React from 'react'
import {
  Dimensions,
  Image
} from 'react-native'
import ImageZoom from 'react-native-image-pan-zoom'
import PropTypes from 'prop-types';

class ZoomableImage extends React.Component {
  constructor(props) {
    super(props)
     this.state = {
       height: 0,
       width: 0
     }
  }

  componentWillMount() {
    const deviceWidth = Dimensions.get('window').width
    const deviceHeight = Dimensions.get('window').height

    Image.getSize(this.props.source.uri, (width, height) => {
      const aspectRatio = Math.min(deviceWidth / width, deviceHeight / height)
      const resizedHeight = height * aspectRatio
      const resizedWidth = width * aspectRatio
      this.setState({ height: resizedHeight, width: resizedWidth })
    })
  }

  render() {
    return (
      <ImageZoom
        cropWidth={Dimensions.get('window').width}
        cropHeight={Dimensions.get('window').height}
        imageWidth={this.state.width}
        imageHeight={this.state.height}
        panToMove={ this.props.allowPanAndZoom }
        pinchToZoom={ this.props.allowPanAndZoom }
        onLongPress={this.props.handlePress}
        longPressTime={ 300 }>
        <Image
          source={ this.props.source }
          style={{width: this.state.width, height: this.state.height}} />
      </ImageZoom>
    )
  }
}


ZoomableImage.propTypes = {
  source: PropTypes.shape({
    uri: PropTypes.string,
  }),
  handlePress: PropTypes.func,
  imageWidth: PropTypes.number,
  imageHeight: PropTypes.number,
  allowPanAndZoom: PropTypes.bool,
}

export default ZoomableImage
