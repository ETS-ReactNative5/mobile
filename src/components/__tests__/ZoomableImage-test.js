import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { ZoomableImage } from '../ZoomableImage'

jest.mock('Dimensions')

const imageSource = {
  uri: 'https://placekitten.com/200/300'
}

it('renders correctly', () => {
  const tree = renderer.create(
    <ZoomableImage
      images={{}}
      source={imageSource}
      handlePress={jest.fn}
      imageWidth={100}
      imageHeight={200}
      allowPanAndZoom={true}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
