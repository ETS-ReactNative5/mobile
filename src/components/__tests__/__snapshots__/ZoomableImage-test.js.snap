// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ImageZoom
  cropHeight={1334}
  cropWidth={750}
  imageHeight={1259}
  imageWidth={675}
  longPressTime={300}
  onLongPress={[Function]}
  panToMove={true}
  pinchToZoom={true}
>
  <Image
    resizeMethod="resize"
    resizeMode="contain"
    source={
      Object {
        "uri": "https://placekitten.com/200/300",
      }
    }
    style={
      Object {
        "height": 1259,
        "width": 675,
      }
    }
  />
</ImageZoom>
`;
